lib/state/wallet/logic.dart

```diff
@@ -1793,4 +1794,54 @@ class WalletLogic extends WidgetsBindingObserver {
       launchUrl(Uri.parse(uri), mode: LaunchMode.externalApplication);
     } catch (_) {}
   }
+
+  void setWalletActionsLoading(bool loading) {
+    _state.setWalletActionsLoading(loading);
+  }
+
+  Future<void> evaluateWalletActions() async {
+    _state.setWalletActionsLoading(true);
+
+    _state.walletActions = [];
+
+    List<ActionButton> actionsToAdd = [];
+
```

Comment:
Just a general comment here, selectors are meant to be a result of state. State is something that tells the UI what to render.

So, I wouldn't rely on checking those properties here. Rather pass them in from the UI if needed. 

But in the 3 cases below, there's a solution for all of them.